### PR TITLE
Feature/booking

### LINKED ISSUE #4: Iteration 2: Traveler Trip Request
As a traveler:

- [ ]  I should be able to make a trip request:
- [ ] I will select a date, duration, number of travelers and choose from a list of destinations
- [ ] After making these selections, I should see an estimated cost (with a 10% travel agent fee) for the trip.
- [ ] Once I submit the trip request, it will show on my dashboard as “pending” so that the travel agency can approve or deny it.

### PR DESCRIPTION
**Pull request by srslie**
**Changes**
- Created booking form display and functionality
- Display confirmation of booking
- Post and Get data following a new booking to stay consistent in the data model
- Added some scss styling
- Further outlined the destinations adding form

**Issues**
-  There's still probably a bit of code that should've been done in another branch, sorry, I will try to be better at staying on track. Solo projects are harder for that because nobody is relying on a specific iteration timeline, but I need to be better at focusing my commits to a specific branch as previously mentioned.
- Some CSS is still not where I want it to be, and I need to refactor some of those mixins.
- Having a harder time getting the accessibility audit to fire on the page with the login.
- Probably could be less wasteful in recalling all users data each time, but I think doing it this way is fine when there's such a small data set, and it will enable dryer code with the agent class.

- [x] DRY & SRP
- [x] Tested Locally
- [x] Accessibility Audited

**Notes**
- I played around with Promise.all for the load of the page, both inside of a 'load' function and out, but I ended up running into weirdness being able to use my getData() function and due to time constraints, I abandoned that direction. I'm using some promises resolutions after the Post and Getting new data to be consistent with the data model, so I'm confident I can deploy those methods if I had to elsewhere.
- To test ```use 'traveler(whateveruseridbetween1-50) and any password``` and then click the book a trip button if you want to see the form.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -21,79 +22,111 @@ function addEvent(area, eventType, func) {
...
-  if (user) {
+
+  const username = document.querySelector('.username').value
+  if (username === 'agent') {
+    user = new Agent
+    domUpdates.toggle(['.login', '.traveler', '.display-booking-button'])
+
+  } else if (username.includes('traveler')) {
+    findUser(username)
     domUpdates.toggle(['.login', '.traveler', '.display-booking-button'])
```
Comment (gaj23) [inline src/index.js:49]: Wicked refactoring of this.

File: src/index.js [hunk 2]
```diff
@@ -132,14 +183,16 @@ function rejectTrip(tripId) {
...
     getData('destinations', destinations)
   ])
-  .then(data => {
-    convertDataIntoClassInstances()
-    displayTrips()
+    .then(() => {
+      user = travelers.find(traveler => traveler.id === userData.id)
+      convertDataIntoClassInstances()
+      displayUserDashboard(user)
+      console.log('WE REFRESH THIS', user.trips)
```
Comment (gaj23) [inline src/index.js:196]: If you wanted user experience, you could try replacing the console.log with an alert. Don't know how well that'll go over with it being called in a promise, but it's an idea.

**Review by gaj23 — APPROVED** [review]
Looks like you're cleaning things up, which is great. I thought this was much more manageable of a branch to view and understand the connection between the different pieces. Thank you for taking my feedback into consideration! 

I think the reason you're having so much trouble running light house is because of the log in page. It was mentioned in the rubric that it was important to run lighthouse before starting work on the logic because of those conflicts. If you have a place that you can go back to, to run that check before you implemented the login page, then that'd be a good back up to have.

File: src/css/_booking-form.scss [hunk 3]
```diff
@@ -14,8 +14,22 @@
...
+  justify-content: center;
+  padding: 3%;
+  align-self: center;
+  width: 80vh;
+  height: 80vh;
+}
+
 .booking-form {
-  // @import formStyle();
+  @include formStyle();
```
Comment (sschipke) [inline src/css/_booking-form.scss:30]: This is cool!

File: src/dom-updates.js [hunk 4]
```diff
@@ -64,28 +72,36 @@ const domUpdates = {
...
+    case ('.upcoming'):
+      return '<h2 class="trips-label">Upcoming Trips</h2>'
+    case ('.pending-trips'):
+      return '<h2 class="trips-label">Pending Trips</h2>'
+    case ('.approved'):
+      return '<h2 class="trips-label">Approved Trips</h2>'
+    case ('.rejected'):
+      return '<h2 class="trips-label">Rejected Trips</h2>'
+    case ('.past'):
+      return '<h2 class="trips-label">Past Trips</h2>'
```
Comment (sschipke) [inline src/dom-updates.js:87]: Love the switch statement!!

**Pull request merged**